### ISSUE TITLE
change the unmber of epochs

### ISSUE THREAD

**Issue by javiergonzalez10upf** [MIC-DKFZ/nnUNet#2272]
Please, I would love to know how can I change the number of epochs. I was not able to do it by reading other related issues. I don't know where should I call the nnUNetTrainer_x_epochs class.

Any help would be useful. Thanks.

**Comment by JackRio** [comment]
If you check the code base you can add a -tr parameter to pass a custom trainer. There are custom trainer located [here](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py) that are suitable for your use case.

`nnUNetv2_train <dataset_name_or_id> <configuration> <fold> nnUNetTrainer_5epochs`
This will run the network for 5 epochs

**Comment by andylitalo** [comment]
I've submitted a PR that allows you to specify any integer number of epochs with an optional flag [here](https://github.com/MIC-DKFZ/nnUNet/pull/2283/files) in case it helps. A little hacky but works :)

**Comment by alarmtran** [comment]
I've changed my code likes you and 
nnUNetv2_train 10 2 2d 0 
This command does not work

**Comment by JackRio** [comment]
@andylitalo I am not sure why you wanted to change the number of epochs but I would prefer other options over mandatory argument to trainer.

1. Implement early stopping if you want to stop training based on validation loss progress.
2. Use a separate trainer for testing your new implementation (Already provided)
3. If you do want to pass it to the trainer I would say make it an optional parameter with a default value of 1000

**Comment by seziegler** [comment]
Change the trainer by providing the `-tr` argument as explained by @JackRio in his earlier message, [here](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/training_length/nnUNetTrainer_Xepochs.py) you can see all available trainer changing the number of epochs.
